flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/StandaloneResourceManagerTest.java

```diff
@@ -83,6 +83,21 @@ public void testNoStartupPeriod() throws Exception {
 		rm.close();
 	}
 
+	@Test
+	public void testStartUpPeriodAfterLeadershipSwitch() throws Exception {
+		final TestingStandaloneResourceManager rm = createResourceManager(Time.milliseconds(1L));
+
+		final Deadline deadline1 = Deadline.fromNow(Duration.ofSeconds(10L));
+		assertHappensUntil(() -> rm.isFailingUnfulfillableRequestAsync().join(), deadline1);
+
+		rm.rmServices.revokeLeadership();
+
+		rm.rmServices.grantLeadership();
+
+		final Deadline deadline3 = Deadline.fromNow(Duration.ofSeconds(10L));
+		assertHappensUntil(() -> rm.isFailingUnfulfillableRequestAsync().join(), deadline3);
```

Comment:
I think the test does not really ensure that we start a new startup period after granting the second leadership. It could also be the case that the timeout won't be reset. I think it would be better to pass in a TestingSlotManager on which we verify the behavior (e.g. that certain methods are being called)